src/components/TextInput/index.js

```diff
@@ -275,12 +276,42 @@ class TextInput extends Component {
     }
   };
 
+  _handleKeyDown = e => {
+    const { onKeyPress } = this.props;
+
+    if (e.which === 8) {
+      onKeyPress({ nativeEvent: { key: 'Backspace' } });
+    }
+  };
+
   _handleKeyPress = e => {
     const { blurOnSubmit, multiline, onKeyPress, onSubmitEditing } = this.props;
     const blurOnSubmitDefault = !multiline;
     const shouldBlurOnSubmit = blurOnSubmit == null ? blurOnSubmitDefault : blurOnSubmit;
     if (onKeyPress) {
-      onKeyPress(e);
+      let keyValue;
+
+      // enter
+      if (e.which === 13) {
+        keyValue = 'Enter';
+      }
+      // space
+      else if (e.which === 32) {
+        keyValue = ' ';
+      }
+      else {
+        // we trim to only care about ones that has a textual representation
+        if (e.shiftKey) {
+          keyValue = String.fromCharCode(e.which).trim();
+        } else {
+          keyValue = String.fromCharCode(e.which).toLowerCase().trim();
```

Comment:
I wonder if it's worth caching the results rather than recomputing each key stroke